spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -449,12 +449,10 @@ class SpdxLicenseWithExceptionExpression(
 
     override fun exception() = exception
 
-    override fun normalize(mapDeprecated: Boolean): SpdxExpression {
+    override fun normalize(mapDeprecated: Boolean) =
         // Manually cast to SpdxLicenseException, because the type resolver does not recognize that in all subclasses of
         // SpdxSimpleExpression normalize() returns an SpdxSingleLicenseExpression.
-        val normalizedLicense = license.normalize(mapDeprecated) as SpdxSingleLicenseExpression
-
-        return when (normalizedLicense) {
+        when (val normalizedLicense = license.normalize(mapDeprecated) as SpdxSingleLicenseExpression) {
```

Comment:
Please also mention in the commit message that this allows turning the function into an expression (which is sort of an incentive to do this inlining).